frb_rust/src/handler.rs

```diff
@@ -241,13 +263,22 @@ impl<EH: ErrorHandler> Executor for ThreadPoolExecutor<EH> {
                         }
                     }
                     Err(error) => {
-                        eh2.handle_error(wrap_info2.port.unwrap(), Error::ResultError(error));
+                        eh2.handle_error(port2, Error::ResultError(error));
                     }
                 };
             });
 
             if let Err(error) = thread_result {
-                eh.handle_error(wrap_info.port.unwrap(), Error::Panic(error));
+                eh.handle_error(port.expect("(worker) eh"), Error::Panic(error));
+            }
+        });
+        #[cfg(not(target_family = "wasm"))]
+        THREAD_POOL.lock().execute(worker);
+
+        #[cfg(target_family = "wasm")]
+        WORKER_POOL.with(|pool| {
+            if let Err(err) = pool.run(worker) {
```

Comment:
For the whole `execute` function: Can we extract THREAD_POOL / WORKER_POOL logic into a separate place? Probably a new struct with a few functions? Then no need to interleave so many `#[cfg]` conditional logic here

(not sure whether I have made similar comments before... The force push seems to confuse GitHub to show previous comments inline in code)

Reply:
There are a few points on WASM that makes this difficult:
- Use of the `transfer!` macro returns different types
- WORKER_POOL is wrapped in a `thread_local!` whereas THREAD_POOL is a top-level static

So even if we create a wrapper struct it's impossible to avoid the inherent complexity of cfg's, which is my rationale for inlining the logic here. I'm open to suggestions, however.

Reply:
Agree. 

For transfer, maybe we can provide the closure (FnOnce) as a generic into the newly created struct and let that struct call `transfer!`.

For thread local vs static, given that both macros can run happily within a function (just like what we have been doing), is it possible to extract to function of struct? Maybe sth like

```rust
impl TheExtractedStruct {
#[cfg(not(target_family = "wasm"))]
  fn pool() {
        lazy_static::lazy_static! {
            static ref THREAD_POOL: parking_lot::Mutex<threadpool::ThreadPool> = ...
        }
    THREAD_POOL
  } 

#[cfg(target_family = "wasm")]
  fn pool() {
        thread_local! {
            static WORKER_POOL: crate::pool::WorkerPool = ...
        }
    WORKER_POOL
  } 
}

// and usage
fn something() {
  let pool = Self::pool();
  pool.do_job();
}